src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectionPool/HttpConnectionPool.Http3.cs

```diff
@@ -87,19 +87,18 @@ internal sealed partial class HttpConnectionPool
                         ThrowGetVersionException(request, 3, reasonException);
                     }
 
-                    long queueStartingTimestamp = HttpTelemetry.Log.IsEnabled() || Settings._metrics!.RequestsQueueDuration.Enabled ? Stopwatch.GetTimestamp() : 0;
-                    Activity? waitForConnectionActivity = ConnectionSetupDistributedTracing.StartWaitForConnectionActivity(authority);
+                    WaitForHttp3ConnectionActivity waitForConnectionTelemetry = new WaitForHttp3ConnectionActivity(Settings._metrics!, authority);
 
                     if (!TryGetPooledHttp3Connection(request, out Http3Connection? connection, out http3ConnectionWaiter))
                     {
+                        waitForConnectionTelemetry.Start();
                         try
                         {
                             connection = await http3ConnectionWaiter.WaitWithCancellationAsync(cancellationToken).ConfigureAwait(false);
                         }
                         catch (Exception ex)
                         {
-                            ConnectionSetupDistributedTracing.ReportError(waitForConnectionActivity, ex);
-                            waitForConnectionActivity?.Stop();
```

Comment:
There is another bug being fixed by this PR: in case of exceptions, we were not reporting the `RequestLeftQueue` EventSource event and the `http.client.request.time_in_queue` metric [like we do in HttpConnectionWaiter](https://github.com/dotnet/runtime/blob/f30fb1576d0dd584e8343ba7610cc9fa31828b7b/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectionPool/HttpConnectionWaiter.cs#L48-L53) for HTTP 1 and 2.

This is now fixed by moving the telemetry stop logic to a shared method.

Reply:
> I couldn't find a way to fail H3 connections in tests.

I haven't tried it, but I think you could simulate connection error with incompatible ALPN:
https://github.com/dotnet/runtime/blob/46dddf167c32ff47af8e235d550f87e87f5ca092/src/libraries/Common/tests/System/Net/Http/Http3LoopbackServer.cs#L21

I.e. creating `Http3LoopbackServer` with options with `Alpn = "not-h3"` or some such.